MecademicRobot/RobotFeedback.py

```diff
@@ -296,30 +318,31 @@ def _get_response_code(self, param):
         else:
             return ['Invalid']
 
-    def _decode_msg(self, response, resp_code):
-        """
-
-        Parameters
-        ----------
-        response : string
-            Message received from the Robot.
-        resp_code : string
-            Message to decode
+    def _convert_response(self, msg):
 
-        Returns
-        --------
-        params : tuplt of float
-            Message decoded.
+        """
+        Method to convert and order the response message received from the robot. It uses regex to find valid
+        response code/messages which are converted and returned into an ordered dict.
 
+        :param msg (str): Entire message received from the robot (with or without the \x00 null character)
+        :return (dict): Contains the response code as keys and the corresponding decoded message as a value.
         """
-        response = response.replace(resp_code+'[','').replace(']','')
-        params = ()
-        if response != '':
-            param_str = response.split(',')
-            if len(param_str) == 6:
-                params = tuple((float(x) for x in param_str))
-            elif len(param_str) == 7:
-                params = tuple((float(x) for x in param_str[1:]))   # remove timestamp
+        # Find the individual responses with a valid format and convert those into a list
+        msg_parts = re.findall("\[....\]\[.+?\]", msg)
+
+        response_dict = {}
+        # Put each individual response message/code in a dictonary
+        for individual_response in msg_parts:
+            response_code = individual_response[1:5]
+            response_msg = individual_response[7:-1].split(',')  # Convert the response message to a list
+
+            if len(response_msg) == 6:
+                response_content = tuple((float(x) for x in response_msg))
+            elif len(response_msg) == 7:
+                response_content = tuple((float(x) for x in response_msg[1:]))   # Remove the timestamp
             else:
-                params =()
-        return params
+                response_content = ()  # If the length of the message is different set the content of to empty
```

Comment:
So if the msg is for instance `[2012][End of Block]` your response_dict will be `{'2012': ()}`?